src/core/address_space.cpp

```diff
@@ -217,14 +218,18 @@ struct AddressSpace::Impl {
 
     void Protect(VAddr virtual_addr, size_t size, bool read, bool write, bool execute) {
         DWORD new_flags{};
-        if (read && write) {
+        if (read && write && execute) {
+            new_flags = PAGE_EXECUTE_READWRITE;
+        } else if (read && write) {
             new_flags = PAGE_READWRITE;
         } else if (read && !write) {
             new_flags = PAGE_READONLY;
-        } else if (!read && !write) {
+        } else if (execute && !read && !write) {
+            new_flags = PAGE_EXECUTE;
+        } else if (!read && !write && !execute) {
             new_flags = PAGE_NOACCESS;
         } else {
-            UNIMPLEMENTED_MSG("Protection flag combination read={} write={}", read, write);
+            LOG_CRITICAL(Common_Memory, "Unsupported protection flag combination");
         }
 
         const VAddr virtual_end = virtual_addr + size;
```

Comment:
Can you remove this loop and simply call VirtualProtect for region specified?

Reply:
Sorry for the delay im getting this error when only calling virtualProtect:
[Common.Memory] <Critical> address_space.cpp:Protect:237: Failed to change virtual memory protect rules
![protect](https://github.com/user-attachments/assets/0fba7790-24ea-4c64-8dfd-02a1cfe8a72a)